express/tests/lib/fixedOTP.test.ts

```diff
@@ -0,0 +1,406 @@
+            expectException = true;
+            respondToMFAChallengeForFixedOTPCredential("testuser1@unit-testing.com", "999999");
+        } catch (error) {
+            checkResult = false;
+
+            if (error instanceof CodeMismatchException) {
+                checkResult = expectException;
+            }
+        }
+
+        expect(checkResult).toBe(true);
+    });
+
+    it("Check isFixedOTPCredential", () => {
+        expect(isFixedOTPCredential("testuser1@unit-testing.com")).toBe(false);
+        expect(isFixedOTPCredential("notestuser1@unit-testing.com")).toBe(false);
+    });
+
+    it("Check isPseudonymisedFixedOTPCredential", () => {
+        expect(isPseudonymisedFixedOTPCredential("~IsValid")).toBe(false);
+        expect(isPseudonymisedFixedOTPCredential("InValid")).toBe(false);
+    });
+
+    it("Check getFixedOTPCredentialEmailAddress", () => {
+        expect(getFixedOTPCredentialEmailAddress("~isValid")).toEqual("");
+        expect(getFixedOTPCredentialEmailAddress("isValid")).toEqual("");
+    });
+
+    it("Check getFixedOTPCredentialTemporaryPassword", () => {
+        expect(getFixedOTPCredentialTemporaryPassword("~isValid")).toEqual("");
+        expect(getFixedOTPCredentialTemporaryPassword("isValid")).toEqual("");
+    });
+
+    it("Check getFixedOTPCredentialSecurityCode", () => {
+        expect(getFixedOTPCredentialSecurityCode("~isValid")).toEqual("");
+        expect(getFixedOTPCredentialSecurityCode("isValid")).toEqual("");
+    });
+
+    it("Check getFixedOTPCredentialMobileNumber", () => {
+        expect(getFixedOTPCredentialMobileNumber("~isValid")).toEqual("");
+        expect(getFixedOTPCredentialMobileNumber("isValid")).toEqual("");
+    });
+
+    it("Check getFixedOTPCredentialSMSCode", () => {
+        expect(getFixedOTPCredentialSMSCode("~isValid")).toEqual("");
+        expect(getFixedOTPCredentialSMSCode("isValid")).toEqual("");
+    });
+});
+
+describe("Check when fixedOTP is active but no Credentials", () => {
+    beforeAll(() => {
+        process.env.USE_STUB_OTP = "true";
+        process.env.FIXED_OTP_CREDENTIALS = "";
+
+        fixedOTPInitialise();
+    });
+
+    it("Check verifyMobileUsingOTPCode", () => {
+        let expectException = false;
+        let checkResult = true;
+
+        try {
+            // Valid SMS Code
+            expectException = true;
+            verifyMobileUsingOTPCode("testuser1@unit-testing.com", "789012");
+
+            // Invalid SMS Code
+            expectException = true;
+            verifyMobileUsingOTPCode("testuser1@unit-testing.com", "123456");
+
+            // Expired SMS Code
+            expectException = true;
+            verifyMobileUsingOTPCode("testuser1@unit-testing.com", "999999");
+        } catch (error) {
+            checkResult = false;
+
+            if (error instanceof CodeMismatchException) {
+                checkResult = expectException;
+            }
+        }
+
+        expect(checkResult).toBe(true);
```

Comment:
Same here as below, these tests pass regardless of whether `verifyMobileUsingOTPCode` is called

Reply:
Yes because the Exception is only raised when the Method is called with an Invalid Value so if you don't call anything the Tests will pass but that is not a Valid Unit Test!

Reply:
Ah I get what you mean. There is a test matcher here for error throwing https://jestjs.io/docs/expect#tothrowerror, it does look a bit weird but that's just javascript weirdness 😄 . You could do something like this to test the error throwing more directly:

```
expect(() =>  verifyMobileUsingOTPCode("testuser1@unit-testing.com", "123456")).toThrow(CodeMismatchException)
```

This way the test fails if there is no exception thrown